### PR TITLE
Make editor big where needed

### LINKED ISSUE #7799: Single CodeMirror instances are too small
# Issue Description
Following the changes in #7790, all CodeMirror instances are rather small. This isn't an issue on a page which has multiple instances, such as the request overview page. However, for the package file page, which has a single CodeMirror instance, this is quite annoying.

Expected Result
============
I expect a file to be completely displayed in a CodeMirror instance when it's the only one on a page.

How to Reproduce
===============

1. Be part of the beta program
2. Go to a page which has only one CodeMirror instance (example: https://build.opensuse.org/package/view_file/openSUSE:Factory/helm-mirror/helm-mirror.spec?expand=1)
3. See small CodeMirror instance

Further Information
===============

Bento:
Screenshot_2019-06-20 openSUSE Build Service

Bootstrap:
bootstrap

### PR DESCRIPTION
**Pull request by eduardoj**
Commit taken from @hellcp 's #7924 pull request.

Fixes #7799.

### CODE REVIEW

**Comment by vpereira** [comment]
@coolo do you want to take a look and see if it's solving your issue?

**Comment by coolo** [comment]
https://obs-reviewlab.opensuse.org/eduardoj-make_editor_bigger_where_needed works great - but is bento only :)

**Comment by eduardoj** [comment]
> https://obs-reviewlab.opensuse.org/eduardoj-make_editor_bigger_where_needed works great - but is bento only :)

Not now :)

**Comment by danidoni** [comment]
> nitpick: I would just change the `big_editor` name for something more technical i.e: `fullscreen_editor` or `no_scrollbar_editor`

Are we using the big editor only for showing files (not editing)? If that's the case, why not using `view_only_editor`?

**Comment by coolo** [comment]
> > nitpick: I would just change the `big_editor` name for something more technical i.e: `fullscreen_editor` or `no_scrollbar_editor`
> 
> Are we using the big editor only for showing files (not editing)? If that's the case, why not using `view_only_editor`?

Because there are other view only editors in requests that should not be big

**Comment by coolo** [comment]
https://obs-reviewlab.opensuse.org/eduardoj-make_editor_bigger_where_needed/package/view_file/home:Admin/ruby/somefile.txt?expand=1 fails my tests.

Going there and scrolling with the mouse still scrolls only milimeters, but not the text. And Ctrl-F won't find you the 'Hidden' I've hidden at the end of the text.

**Comment by coolo** [comment]
if I move the `height: auto` to the .CodeMirror it works. On the div it's toothless

**Comment by vpereira** [comment]
> if I move the `height: auto` to the .CodeMirror it works. On the div it's toothless

won't that reintroduce that https://github.com/openSUSE/open-build-service/commit/a386a035108da5e4360cdd472b103174337b8516 ?

**Comment by coolo** [comment]
https://build.opensuse.org/projects/openSUSE:Leap:15.2/meta is affected as well, btw

> > if I move the `height: auto` to the .CodeMirror it works. On the div it's toothless
> 
> won't that reintroduce [a386a03](https://github.com/openSUSE/open-build-service/commit/a386a035108da5e4360cdd472b103174337b8516) ?

That performance problem is about requests/show - let's worry about that one afterwards.

**Comment by hellcp** [comment]
Yeah, a better way to approach this was to probably keep that `height:auto` and add a class for `height: 300px`, because that's only used once or twice in all the views. Big editor appears everywhere but the pages that include multiple editors, which are review and diff views.

**Comment by hennevogel** [comment]
Just don't use `big_editor: true` on `BsRequest#show`...

**Comment by eduardoj** [comment]
The `height: auto` was moved to the` .CodeMirror`. @hennevogel, @hellcp, @coolo, @vpereira, @danidoni, would you mind taking another look?

File: src/api/app/views/webui2/shared/_editor.html.haml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 :ruby
   save ||= {}
-  style ||= { read_only: false }
+  style ||= { read_only: false, big_editor: false }
```
Comment (hennevogel) [inline src/api/app/views/webui2/shared/_editor.html.haml]: Hm, we only have one page where this should be false, `BsRequest#show`. How about turning this default around?

**Comment by coolo** [comment]
https://obs-reviewlab.opensuse.org/eduardoj-make_editor_bigger_where_needed/package/view_file/home:Admin/ruby/somefile.txt?expand=1 works now.

And I think I see a case of having the Save button at the top there :)

File: src/api/app/views/webui2/shared/_editor.html.haml (hunk 1 above)
Comment (eduardoj) [reply src/api/app/views/webui2/shared/_editor.html.haml]: I agree. I will change the default.

Comment (eduardoj) [reply src/api/app/views/webui2/shared/_editor.html.haml]: Done. Should I merge all the commits into one?

Comment (hennevogel) [reply src/api/app/views/webui2/shared/_editor.html.haml]: Yeah sounds like a good idea :)

**Pull request merged**